pkg/services/validator/aws.go

```diff
@@ -714,6 +717,23 @@ func readAwsCredentials(c *components.AWSPluginConfig, tc *cfg.TaskConfig, k8sCl
 		if err != nil {
 			return err
 		}
+
+		if tc.Direct {
```

Comment:
If we do end up prompting for for AWS creds in direct mode and the user provides STS details, we need to actually obtain temporary creds via the STS SDK and set those values as the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` env vars.

Reply:
Isn't this handled by the plugin, regardless of mode? I thought we just need to pass the STS auth details to the plugin directly?

Reply:
My bad. You're right. It's handled in `NewAPI`. You just have to construct a proper `v1alpha1.AwsAuth` and pass it in.